apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -102,6 +106,20 @@
       this.initService();
     },
 
+    register(appInstanceId) {
+      var mobileMessageClient = MozMobileMessageClient.forApp(appInstanceId);
+
+      MESSAGE_EVENTS.forEach((event) => {
+        mobileMessageClient.on(event, (data) => {
+          // Workaround to avoid the duplicated events by inserting
+          // appInstanceId into event name in bridge mixin.
+          data.appInstanceId = appInstanceId;
+          this.broadcast(event, data);
+        });
+      });
+      return;
+    },
```

Comment:
also I think we should send back the full affected message together with the event.